app/src/main/java/org/ole/planet/myplanet/ui/team/TeamPagerAdapter.kt

```diff
@@ -51,56 +42,26 @@ class TeamPagerAdapter(fm: FragmentActivity, team: RealmMyTeam?, private val isI
 
     @RequiresApi(Build.VERSION_CODES.O)
     override fun createFragment(position: Int): Fragment {
-        val f: Fragment? = if (!isInMyTeam) {
-            if (position == 0) PlanFragment() else {
-                JoinedMemberFragment()
-            }
-        } else {
-            checkCondition(position)
+        if (position < 0 || position >= list.size) {
+            throw IllegalArgumentException("Invalid position: $position. List size: ${list.size}")
         }
-        val b = Bundle()
-        b.putString("id", teamId)
-        f!!.arguments = b
-        return f
-    }
-
-    @RequiresApi(Build.VERSION_CODES.O)
-    private fun checkCondition(position: Int): Fragment? {
-        var f: Fragment? = null
-        when (position) {
-            0 -> f = DiscussionListFragment()
-            1 -> f = PlanFragment()
-            2 -> f = JoinedMemberFragment()
-            3 -> f = TeamTaskFragment()
-            4 -> f = EnterpriseCalendarFragment()
-            5 -> f = fragment //finances
-            6 -> {
-                if (isEnterprise) {
-                    f = ReportsFragment()
-                } else {
-                    f = TeamResourceFragment()
-                    MainApplication.listener = f
-                }
-            }
-            7 -> {
-                if (isEnterprise) {
-                    f = TeamResourceFragment()
-                    MainApplication.listener = f
-                } else {
-                    f = MembersFragment()
-                }
-            }
-            8 -> {
-                if (isEnterprise) {
```

Comment:
we are also missing the finances data in enterprises. in the video below you'll see it assumes the finances tab is the team courses tab
[Screen_recording_20240710_192253.webm](https://github.com/open-learning-exchange/myplanet/assets/64019708/bfe5365e-524f-47b1-b431-6f6804a5899d)

this is a video of how it should be
[Screen_recording_20240710_192622.webm](https://github.com/open-learning-exchange/myplanet/assets/64019708/fd4a702c-05af-4f76-ad0c-b657c6dc71bb)